src/fu-engine.c

```diff
@@ -2926,7 +2926,11 @@ fu_engine_plugin_device_added_cb (FuPlugin *plugin,
 				  gpointer user_data)
 {
 	FuEngine *self = (FuEngine *) user_data;
-	fu_device_set_priority (device, fu_plugin_get_priority (plugin));
+	gint priority = fu_plugin_get_priority (plugin);
+	GPtrArray *children = fu_device_get_children (device);
+	if (children->len > 0)
+		priority ++;
```

Comment:
I think this is too simple; you want the parent priority to be *1 more* than the biggest child priority, not just one more than the plugin priority. i.e. we'd still get `fu_plugin_get_priority (plugin)` first, and then increase it further if it's not larger than all the child priorities.

Reply:
Setting priority is a private function that plugins can't use. So a simple routine made sense to me in that sense.

Reply:
What happens if the parent plugin priority is 1, but the child plugin priority is 3?